pkg/tool/osdctl/osdctl.go

```diff
@@ -0,0 +1,191 @@
+package osdctl
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	gogithub "github.com/google/go-github/github"
+	"github.com/openshift/backplane-tools/pkg/source/github"
+)
+
+// Tool implements the interface to manage the 'osdctl' binary
+type Tool struct {
+	source *github.Source
+}
+
+func NewTool() *Tool {
+	t := &Tool{
+		source: github.NewGithubSource("openshift", "osdctl"),
+	}
+	return t
+}
+
+func (t *Tool) Name() string {
+	return "osdctl"
+}
+
+func (t *Tool) Install(rootDir, latestDir string) error {
+	// Pull latest release from GH
+	release, err := t.source.FetchLatestRelease()
+	if err != nil {
+		return err
+	}
+
+	// Determine which assets to download
+	var checksumAsset gogithub.ReleaseAsset
+	var osdctlBinaryAsset gogithub.ReleaseAsset
+	var arch = runtime.GOARCH
+	if arch == "amd64" {
+		arch = "x86_64"
+	}
+	for _, asset := range release.Assets {
+		fmt.Println(asset.GetName())
+
+		if asset.GetName() == "sha256sum.txt" {
+			if checksumAsset.GetName() != "" {
+				return fmt.Errorf("detected duplicate osdctl checksum assets")
+			}
+			checksumAsset = asset
+			if osdctlBinaryAsset.GetName() != "" {
+				return fmt.Errorf("detected duplicate osdctl binary assets")
+			}
+			osdctlBinaryAsset = asset
+			continue
+		}
```

Comment:
This loop will assign both `osdctlBinaryAsset` and `checksumAsset` to the same `sha256sum.txt` file. The `osdctlBinaryAsset` should be discovered similar to [how `ocm` does it](https://github.com/tnierman/backplane-tools/blob/main/pkg/tool/ocm/ocm.go#L41-L62)